assets/sass/_pulumi-up.scss

```diff
@@ -0,0 +1,100 @@
+// Neon Animation found on Codepen
+// https://codepen.io/Trinca/pen/NAvpWa
+$pageBackgroundColor: #11151c;
+
+.neon-container {
+  text-align: center;
+  vertical-align: middle;
+}
+
+@mixin neon-sign-defaults($fontSize: 4rem) {
+  font-family: "Vibur";
+  font-size: $fontSize;
+}
+
+@mixin neon-sign-animation($light, $medium, $dark, $pulseTime, $name) {
+    animation: $name $pulseTime linear infinite;
+    -moz-animation: $name $pulseTime linear infinite;
+    -webkit-animation: $name $pulseTime linear infinite;
+    -o-animation: $name $pulseTime linear infinite;
+
+    @keyframes #{$name} {
+      0%,
+      100% {
+        text-shadow: 0 -40px 100px, 0 0 2px, 0 0 1em $medium, 0 0 0.5em $medium, 0 0 0.1em $medium, 0 10px 3px #000;
+        color: $light;
+      }
+      50% {
+        text-shadow: 0 -40px 100px, 0 0 2px, 0 0 1em $dark, 0 0 0.5em $dark, 0 0 0.1em $dark, 0 10px 3px #000;
+        color: $medium;
+      }
+    }
+}
+
+@mixin neon-sign-purple($fontSize, $name: "neon-sign-purple-pulse", $animate: true) {
+  @include neon-sign-defaults($fontSize);
+  @if $animate {
+    @include neon-sign-animation(theme('colors.purple.300'), theme('colors.purple.600'), theme('colors.purple.800'), 4s, $name);
+  }
+
+  color: theme('colors.purple.400');
+  text-shadow: 0 -40px 100px, 0 0 2px, 0 0 1em theme('colors.purple.700'), 0 0 0.5em theme('colors.purple.700'), 0 0 0.1em theme('colors.purple.700'), 0 10px 3px #000;
+}
+
+@mixin neon-sign-green($fontSize, $name: "neon-sign-green-pulse", $animate: true) {
+  @include neon-sign-defaults($fontSize);
+  @if $animate {
+    @include neon-sign-animation(theme('colors.green.400'), theme('colors.green.700'), theme('colors.green.900'), 3s, $name);
+  }
+
+  color: theme('colors.blue.400');
+  text-shadow: 0 -40px 100px, 0 0 2px, 0 0 1em theme('colors.green.700'), 0 0 0.5em theme('colors.green.700'), 0 0 0.1em theme('colors.green.700'), 0 10px 3px #000;
```

Comment:
I suspect we could reduce a lot of duplication here by collapsing these into a single parameterized mixin that accepts colors and uses a Sass map, along with a handful of local variables.

Would you mind also adding a few comments explaining what all of this is doing? I can pretty much tell, but there's kind of a lot of going on here.

Reply:
👍